### ISSUE TITLE
caffe and pytorch result is not same 

### ISSUE THREAD

**Issue by gmt710** [JDAI-CV/fast-reid#210]
Hi, I followed your deploy steps . but also get different result. I focused on this for several weeks. 
Can you help me?

**Comment by L1aoXingyu** [comment]
It's normal when caffe outputs and pytorch outputs are different. In my project, the cosine similarity of caffe outputs and pytorch outputs is from 0.95 to 0.99.

**Comment by gmt710** [comment]
Firstly, Greatly thanks for your reply. 
Yes, But when people wear similar clothes, the impact may be greater.
I'm verifying that if there's progress, I'll push it forward.

I am curious about the yaml config of your sharing weight(like bagtricks). Because I can't get the great result like you.

**Comment by L1aoXingyu** [comment]
I just train the config with a single GPU and don't modify anything. What result did you get?

**Comment by gmt710** [comment]
I also use the original bagtricks_R50 config file with a single GPU, but I get the best mAP 85%.
Even if I modify the config file(adding circle loss as so on), I can't get mAP 86.1%. So I wonder if some places are set up incorrectly.

**Comment by L1aoXingyu** [comment]
![image](https://user-images.githubusercontent.com/19245504/89378196-e5f4aa00-d725-11ea-8009-c913f7f3b3d9.png)

I just rerun and can get this result.

**Comment by gmt710** [comment]
Nice.I get two good news:
1.pytorch to caffe ,get Similar results.modify this:
```
# self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
self.maxpool = nn.MaxPool2d(kernel_size=3, stride=2, ceil_mode=True)
```
2.retrain by this setting, I get this result:
![image](https://user-images.githubusercontent.com/34018741/89379612-aa0f1400-d728-11ea-8caf-a9cf786b163e.png)

I have a new problem. I test my model on my own data and find ID mismatch when people wear similar clothes.

**Comment by L1aoXingyu** [comment]
Good news! I may change the `maxpooling` for adapting to caffe converter. 

The similar clothes is always a hard problem for person re-id, because it heavily depends on appearance. You can consider using face recognition to identify people wearing similar clothes.

**Comment by gmt710** [comment]
Thanks for your reply. I hope we can discuss it together in the future.
By the way, I create a PR.

**Comment by L1aoXingyu** [comment]
Thanks for your PR and I need to do some experiments and then merge your PR.

**Comment by mengban** [comment]
I  have a question about torch->caffe model. I saw in your blog, u succeed ! What's my question is that can the convert script convert any pytorch model to caffe, or it just work sfor fast-reid? @gmt710

**Comment by gmt710** [comment]
This code is similar to https://github.com/xxradon/PytorchToCaffe. 
So I think the convert script convert any pytorch model to caffe.

**Comment by mengban** [comment]
thank u. It may help me a lot.
